consensus/cryptoiface.go

```diff
@@ -0,0 +1,82 @@
+package consensus
+
+import (
+	"github.com/relab/hotstuff"
+)
+
+// CryptoBase provides the basic cryptographic methods needed to create, verify, and combine signatures.
+type CryptoBase interface {
+	// Sign creates a cryptographic signature of the given message.
+	Sign(message []byte) (signature QuorumSignature, err error)
+	// Verify verifies the given cryptographic signature according to the specified options.
+	// NOTE: One of either VerifySingle or VerifyMulti options MUST be specified,
+	// otherwise this function will have nothing to verify the signature against.
+	Verify(signature QuorumSignature, options ...VerifyOption) bool
+	// Combine combines multiple signatures into a single signature.
+	Combine(signatures ...QuorumSignature) (signature QuorumSignature, err error)
+}
+
+// VerifyOption sets options for the Verify function in the CryptoBase interface.
+type VerifyOption func(o *VerifyOptions)
+
+// VerifyThreshold specifies the number of signers that must have participated
+// in the signing of the signature in order for Verify to return true.
+func VerifyThreshold(threshold int) VerifyOption {
+	return func(o *VerifyOptions) {
+		o.Threshold = threshold
+	}
+}
+
+// VerifySingle verifies the signature against the given message.
+//
+// NOTE: VerifySingle and VerifyMulti cannot be used together,
+// and whichever option is specified last will be the effective one.
```

Comment:
If they cannot be used together, would it not be better to panic, since it would be a programmer mistake to combine them??